handler.go

```diff
@@ -73,23 +74,39 @@ func (handler *Handler) ServeHTTP(writer http.ResponseWriter, req *http.Request)
 			handler.dismiss(body)
 
 			requestCount = requestCount + 1
-			if response.Err != nil {
-				responses[backendName] = response
-			}
+			responses[backendName] = response
 
 			if requestCount >= backendCount {
 				if handler.server.onBackendFinishedHandler != nil {
 					handler.server.onBackendFinishedHandler(responses)
 				}
 
+				resp := make([]*Response, 0, len(responses))
+				for _, r := range responses {
+					resp = append(resp, r)
+				}
+
+				allResponsesCh <- resp
+
 				done <- true
```

Comment:
Could we reorganize the code a bit to make it more readable? Move everything, except for the `break` statement of course, outside the `for` loop.